### PR TITLE
Concf 63 go search

### PR DESCRIPTION
**Pull request by justnpT**
developed based on https://wikia-inc.atlassian.net/browse/CONCF-63#

Adding 3 goSearch Tests

Make sure go search is turned off for anon	 
Make sure go search is turned off for logged in	 
Make sure go search is turned on for logged in with preference	 

Make sure go search is activated when choosing a suggestion for anon 
Make sure go search is activated when choosing a suggestion for user

### CODE REVIEW

**Comment by justnpT** [comment]
closed for it requires a little bit more work for stability purposes

**Comment by justnpT** [comment]
stabilized, good to CR

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/global_navitagtion/NavigationBar.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	public void triggerSuggestions(String query) {
+	    searchInput.click();
+	    waitForElementByBy(jqueryAutocompleteBy);
+	    sendKeysHumanSpeed(searchInput, query);
+	    waitForElementByElement(suggestion);		
+	    waitForElementByElement(suggestionsList.get(0));		
+	}
+	
+	public void verifySuggestions(String suggestion) {
+		waitForElementByElement(suggestionsList.get(0));
```
Comment (Dyktus) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/global_navitagtion/NavigationBar.java]: This indentation is too big.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/global_navitagtion/NavigationBar.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+		}
+		Assertion.assertStringContains(suggestion, allSuggestions);
+	}
+
+	/**
+    * This method will arrow down through suggestions 
+    * until it finds given suggestion, then it clicks enter.
+	*/
+	public ArticlePageObject ArrowDownAndEnterSuggestion(String suggestion) {
+		waitForElementByElement(suggestionsList.get(0));
```
Comment (Dyktus) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/global_navitagtion/NavigationBar.java]: This indentation can be smaller.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/search/intrawikisearch/IntraWikiSearchPageObject.java [hunk 3]
```diff
@@ -111,16 +111,20 @@ public void searchFor(String query) {
...
+  public void verifySuggestions(String suggestion) {
     waitForElementByElement(suggestionsList.get(0));
     for (int i = 0; i < suggestionsList.size(); i++) {
-      Assertion.assertStringContains(query, suggestionsList.get(i).getText());
+      Assertion.assertStringContains(suggestion, suggestionsList.get(i).getText());
     }
   }
 
+  public void triggerSuggestions(String query) {
+	    searchField.click();
```
Comment (Dyktus) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/search/intrawikisearch/IntraWikiSearchPageObject.java]: Indentation.

File: src/test/java/com/wikia/webdriver/testcases/searchtests/intrawiki/BasicActions.java [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+ *  11. Verify if there are correct advanced option set as a default
+ *  12. Search for some image without typing extension (.jpg) and verify photo is found
+ *  13. Search for different phrases and verify there are correct namespaces in result titles
+ *  14. Search for empty field and verify if search page is opened
+ *  15. Verify top module
+ *  16. Verify push to top is working in community.wikia.com
+ */
+
+/**
+ * @author
```
Comment (Dyktus) [inline src/test/java/com/wikia/webdriver/testcases/searchtests/intrawiki/BasicActions.java]: Who is the author of this code ? :)

File: src/test/java/com/wikia/webdriver/testcases/searchtests/intrawiki/goSearchFeature.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+*  AnonFromSuggestion: Make sure clicking enter on suggestion takes you to destination page for anonymous
+*  UserFromSuggestion: Make sure clicking enter on suggestion takes you to destination page for logged in user 
+*  goSearchPreference: Make sure clicking search button after typing article name to the search field takes you to destination page when user has the preference enabled
+*/
+
+public class goSearchFeature extends BasicActions {
+	
+	@Test(groups = {"Search", "IntraWikiSearch", "goSearchFeature", "AnonFromSuggestion"})
+	public void AnonFromSuggestion() {
+		WikiBasePageObject base = new WikiBasePageObject(driver);
```
Comment (Dyktus) [inline src/test/java/com/wikia/webdriver/testcases/searchtests/intrawiki/goSearchFeature.java]: I'm curious why there is bigger indentation ? Look also to next method.

**Comment by justnpT** [comment]
looks LGTM to me

File: src/test/java/com/wikia/webdriver/testcases/searchtests/intrawiki/Suggestions.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import com.wikia.webdriver.common.contentpatterns.SearchContent;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.search.intrawikisearch.IntraWikiSearchPageObject;
+
+/*
+*  dropDownSuggestions: Type at least 3 chars and verify suggestions are displaying and contain given phrase
+*  searchSuggestionsVisibility: Verify Image and text is shown next to every suggestion
+*/
+
+public class Suggestions extends BasicActions {
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/testcases/searchtests/intrawiki/Suggestions.java]: please don't extend class with tests

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/search/intrawikisearch/IntraWikiSearchPageObject.java [hunk 7]
```diff
@@ -21,343 +21,349 @@
 
 public class IntraWikiSearchPageObject extends SearchPageObject {
 
-  private String photoExtension = ".jpg";
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/search/intrawikisearch/IntraWikiSearchPageObject.java]: use final

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/search/intrawikisearch/IntraWikiSearchPageObject.java [hunk 8]
```diff
@@ -21,343 +21,349 @@
...
-  @FindBy(css = "#WikiaSearchHeader .search-suggest li:not(.all)")
-  private List<WebElement> newSuggestionsList;
-  @FindBy(css = ".block")
-  private List<WebElement> suggestionTextsList;
-  @FindBy(id = "searchInput")
-  private WebElement searchInputInGlobalNav;
-  @FindBy(id = "searchForm")
-  private WebElement searchFormInGlobalNav;
-
-  private By jqueryAutocompleteBy = By.cssSelector("[src*='jquery.autocomplete']");
```
Comment (ludwikkazmierczak) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/search/intrawikisearch/IntraWikiSearchPageObject.java]: Can be final

**Comment by justnpT** [comment]
looks LGTM for me now. I will merge it in the evening if not objections

**Pull request merged**